lucet-runtime/lucet-runtime-internals/src/future.rs

```diff
@@ -195,6 +195,9 @@ impl InstanceHandle {
                             ))
                         }
                     }
+                    RunResult::BoundExpired => {
+                        panic!("BoundExpired was not expected when no bound was set");
```

Comment:
I think here what you'll probably want to do is to add a new variant to `BounceInner` like `BounceInner::Pause`. Then below in the `match bounce` when it gets `Pause` you'll want to do something like [`tokio::task::yield_now`](https://docs.rs/tokio/0.2.9/tokio/task/fn.yield_now.html).

Reply:
Ah I see though this will probably also need a new argument to `run_async` to specify the bound itself!

Reply:
This worked, thanks!